src/portfft/descriptor.hpp

```diff
@@ -542,67 +485,110 @@ class committed_descriptor {
+
+      if (is_global) {
+        const bool is_final_factor = counter == (prepared_vec.size() - 1);
+
+        length = std::accumulate(factors.begin(), factors.end(), Idx(1), std::multiplies<Idx>());
+
+        remaining_factors_prod /= length;
+        forward_stride = remaining_factors_prod;
+        backward_stride = remaining_factors_prod;
+        forward_distance = is_final_factor ? length : 1;
+        backward_distance = is_final_factor ? length : 1;
+
+        if (is_final_factor) {
+          multiply_on_store = detail::elementwise_multiply::NOT_APPLIED;
+          apply_scale = detail::apply_scale_factor::APPLIED;
           if (compute_direction == direction::BACKWARD) {
             conjugate_on_store = detail::complex_conjugate::APPLIED;
+          } else {
+            conjugate_on_store = detail::complex_conjugate::NOT_APPLIED;
           }
-          set_spec_constants(detail::level::GLOBAL, in_bundle, factor_size, factors,
-                             detail::elementwise_multiply::NOT_APPLIED, detail::elementwise_multiply::NOT_APPLIED,
-                             detail::apply_scale_factor::APPLIED, level, conjugate_on_load, conjugate_on_store,
-                             scale_factor, static_cast<Idx>(counter), static_cast<Idx>(prepared_vec.size()));
-          // reset conjugate_on_store
-          conjugate_on_store = detail::complex_conjugate::NOT_APPLIED;
+          conjugate_on_load = detail::complex_conjugate::NOT_APPLIED;
         } else {
+          multiply_on_store = detail::elementwise_multiply::APPLIED;
+          apply_scale = detail::apply_scale_factor::NOT_APPLIED;
+          conjugate_on_store = detail::complex_conjugate::NOT_APPLIED;
           if (counter == 0 && compute_direction == direction::BACKWARD) {
             conjugate_on_load = detail::complex_conjugate::APPLIED;
+          } else {
+            conjugate_on_load = detail::complex_conjugate::NOT_APPLIED;
           }
-          set_spec_constants(detail::level::GLOBAL, in_bundle, factor_size, factors,
-                             detail::elementwise_multiply::NOT_APPLIED, detail::elementwise_multiply::APPLIED,
-                             detail::apply_scale_factor::NOT_APPLIED, level, conjugate_on_load, conjugate_on_store,
-                             scale_factor, static_cast<Idx>(counter), static_cast<Idx>(prepared_vec.size()));
-          // reset conjugate_on_load
-          conjugate_on_load = detail::complex_conjugate::NOT_APPLIED;
         }
       } else {
+        length = static_cast<Idx>(params.lengths[dimension_num]);
+
+        forward_stride = static_cast<Idx>(params.forward_strides[dimension_num]);
+        backward_stride = static_cast<Idx>(params.backward_strides[dimension_num]);
+        if (is_multi_dim) {
+          if (dimension_num == (params.lengths.size() - 1)) {
+            forward_distance = length;
+            backward_distance = length;
+            apply_scale = detail::apply_scale_factor::APPLIED;
```

Comment:
I would have a small preference towards the way the it was being decided upon whether or not to skip scaling in the case of multi-dimensional transforms, as the `is_multi_dim` check is not required. It would also reduce the number of cases we need to handle inside `set_spec_const_driver`